File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03042/p03042_num7_parsed.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string S;
    cin >> S;

    if ((S.substr(0, 2) >= "01" && S.substr(0, 2) <= "12") && (S.substr(2, 2) >= "01" && S.substr(2, 2) <= "12")) {
        cout << "AMBIGUOUS" << endl;
    } else if ((S.substr(0, 2) >= "01" && S.substr(0, 2) <= "12") && (S.substr(2, 2) >= "13" && S.substr(2, 2) <= "99")) {
        cout << "MMYY" << endl;
    } else if ((S.substr(0, 2) >= "13" && S.substr(0, 2) <= "99") && (S.substr(2, 2) >= "01" && S.substr(2, 2) <= "12")) {
        cout << "YYMM" << endl;
    } else {
        cout << "NA" << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03042/p03042_num7_parsed.py
# Get the input digit sequence
S = input().strip()

# Check if S is valid in YYMM format
valid_yymm = False
if int(S[:2]) >= 1 and int(S[:2]) <= 12:
    valid_yymm = True

# Check if S is valid in MMYY format
valid_mmyy = False
if int(S[2:]) >= 1 and int(S[2:]) <= 12:
    valid_mmyy = True

# Print the result
if valid_yymm and valid_mmyy:
    print("AMBIGUOUS")
elif valid_yymm:
    print("YYMM")
elif valid_mmyy:
    print("MMYY")
else:
    print("NA")
